File: python/problem-005.py
"""
Copyright 2016 Peter Beard
Distributed under the GNU GPL v2. For full terms, see the LICENSE file.

2520 is the smallest number that can be divided by each of the numbers from 1 to
10 without any remainder.

What is the smallest positive number that is evenly divisible by all of the
numbers from 1 to 20?
"""
def is_divisible(n, factors):
    """Determine whether n is divisible by all of [factors]"""
    for f in factors:
        if n % f != 0:
            return False
    return True


def solution():
    n = 2520
    while True:
        n += 20
        if is_divisible(n, range(20, 10, -1)):
            return n

print("The smallest number divisible by all of [1,20] is {}".format(solution()))

File: c/src/problem-005.c
/*
 * Copyright 2016 Peter Beard
 * Distributed under the GNU GPL v2. For full terms, see the LICENSE file.
 *
 * Problem #5
 *
 * 2520 is the smallest number that can be divided by each of the numbers from 1
 * to 10 without any remainder.
 *
 * What is the smallest positive number that is evenly divisible by all of the
 * numbers from 1 to 20?
 */
#include <stdio.h>

int main()
{
    int n = 20;
    while(1)
    {
        n += 20;
        int i = 20;
        for(i = 20; i > 1; i--)
        {
            if(n % i != 0)
                break;
        }
        if(i == 1)
            goto end;
    }
    end:printf("The smallest number divisible by 1 - 20 is %d\n", n);
}
